hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -182,6 +185,20 @@ public void commitMigration(PartitionMigrationEvent event) {
                     event.getNewReplicaIndex());
         }
 
+        if (SOURCE == event.getMigrationEndpoint()) {
+            PartitionContainer partitionContainer
+                    = mapServiceContext.getPartitionContainer(event.getPartitionId());
+            for (RecordStore recordStore : partitionContainer.getAllRecordStores()) {
+                Set<MapOperation> offloadedOperations = recordStore.getOffloadedOperations();
+                List<MapOperation> opList = new ArrayList<>(offloadedOperations.size());
+                opList.addAll(offloadedOperations);
+                for (MapOperation op : opList) {
+                    PartitionMigratingException pme = createPME(event, op);
+                    op.getOperationResponseHandler().sendResponse(op, pme);
```

Comment:
Could you document how we handle exception which can happen in the step transitions or while running the operation in the offloading thread? Also, what is expected from the operation steps to avoid corrupting the internal state of the system in case of the exceptions.

Reply:
Added docs for `StepSupplier#nextStep`